Automattic/jetpack
# Jetpack AI: Generate a post intro based on title and categories

This change aims to make the "Jetpack AI paragraph"  block at least half-intelligent.

## Proposed changes:

If the block is inserted into an empty editor canvas, it will decide it is used to generate new content.
It will proceed to use Gutenberg core data bindings to grab title and categories it's being fed and generate a post intro based on this information.

When no categories are selected, it will work solely off the title.
It would be neat if Jetpack SEO allowed us to select SEO Focus Keywords - that way we could supply that to AI as well.

Here is content generated based on title and category:
![Zrzut ekranu 2023-01-12 o 15 40 27](https://user-images.githubusercontent.com/3775068/212098300-1c057ba3-fcb5-469c-9cf7-e781be260e9b.png)


## Jetpack product discussion

pbOQVh-2TF-p2

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

1. Get a Jetpack site. You can run locally with ngrok. You don't need to point to your sandbox.
2. Pretend the site is atomic with instructions from https://github.com/Automattic/jetpack/pull/28043#issuecomment-1377546608
3. Enable JETPACK_BETA_BLOCKS
4. Prepare a post. Fill out only title and categories
5. Insert AI Paragraph block into empty canvas and observe it working
